packages/grafana-ui/src/types/icon.ts

```diff
@@ -214,3 +224,18 @@ export function getFieldTypeIcon(field?: Field): IconName {
   }
   return 'question-circle';
 }
+
+function isValidIconName(iconName: string): iconName is IconName {
+  const namedIcons: string[] = availableIcons;
+  const brandIcons: string[] = avaibleBrandIcons;
+
+  return namedIcons.includes(iconName) || brandIcons.includes(iconName);
+}
+
+export function toIconName(iconName: string): IconName | undefined {
```

Comment:
love this - so many `as IconName`s can be removed

Reply:
@joshhunt this would be better if we can validate that the string is indeed an icon Name. Any random string as icon can be problematic as we don't know if those icons actually exists.

Reply:
Is this not what this does?

Reply:
@joshhunt Yes please ignore my confusion I had my  mind set in something else.